cherab/tools/observers/spectroscopy/instrument.py

```diff
@@ -0,0 +1,88 @@
+
+# Copyright 2014-2017 United Kingdom Atomic Energy Authority
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+class SpectroscopicInstrument:
```

Comment:
This base class doesn't subclass `Node` or any of the `Observer` classes, so will never be part of a scenegraph. Presumably this is because each instance will have an observer (probably fibre bundle). This is unlike every other observer in Cherab, which is treated as a scenegraph node in  order to observe anything.

I'd suggest making a note of this in the documentation, as it could cause confusion ("why doesn't `cherab.tools.observers.SpectroscopicInstrument` have a `parent` argument?"). Since its always necessary to have an observer linked to this instrument for it to be of use, why not add an `observer` property too, to save every derived class having to define their own?

Reply:
> Since its always necessary to have an observer linked to this instrument for it to be of use, why not add an `observer` property too, to save every derived class having to define their own?

A single instrument can have multiple observers connected as well as a single observer can be connected to a multiple instruments. Even if we introduce the `observers` property, it's not clear to me how it can be used. For example, to setup the spectral range of the observer we must know the list of all instruments to which this observer is connected. The code sample added to the docstring for `SurveySpectrometer` now shows this case:
```python
from numpy import ceil
from raysect.optical import World
from raysect.optical.observer import FibreOptic
from cherab.tools.spectroscopy import SurveySpectrometer, Polychromator, PolychromatorFilter

world = World()
fibreoptic = FibreOptic(name="MyFibreOptic", parent=world)
# Here, the fibre optic is "connected" to both survey spectrometer and polychromator.

# setting up the polychromator
h_alpha_filter = PolychromatorFilter(656.1, name='H-alpha filter')
ciii_465nm_filter = PolychromatorFilter(464.8, name='CIII 465 nm filter')
polychromator = Polychromator([h_alpha_filter, ciii_465nm_filter], name='MyPolychromator')

# setting up the survey spectrometer
spectrometer = SurveySpectrometer(0.1, 1024, 500, name='MySpectrometer')

fibreoptic.min_wavelength = min(spectrometer.min_wavelength, polychromator.min_wavelength)
fibreoptic.max_wavelength = max(spectrometer.max_wavelength, polychromator.max_wavelength)
bin_width = min((spectrometer.max_wavelength - spectrometer.min_wavelength) / spectrometer.spectral_bins,
                (polychromator.max_wavelength - polychromator.min_wavelength) / polychromator.spectral_bins)
fibreoptic.spectral_bins = int(ceil((fibreoptic.max_wavelength - fibreoptic.min_wavelength) / bin_width))
fibreoptic.pipelines = spectrometer.pipelines() + polychromator.pipelines()
```